### PR TITLE
Address some vulnerabilities reported for ORT using resolutions in .ort.yml

### PR DESCRIPTION
**Pull request by oheger-bosch**
This PR is in the context of scanning ORT with ORT. Sonatype NexusIQ which we use currently as advisor issues a number of vulnerabilities for ORT. In this PR I try to address some of them by resolutions in `.ort.yml` that (according to my analysis and understanding) should not affect ORT in the way it is using the affected components.

There are other vulnerabilities though which probably cannot be handled by such resolutions in general, since they may depend on the concrete context or environment in which ORT is used.

### CODE REVIEW

**Review by sschuberth — APPROVED** [review]
Very nicely crafted resolution comments, @oheger-bosch!

File: .ort.yml [hunk 1]
```diff
@@ -31,6 +31,10 @@ excludes:
     reason: "DEV_DEPENDENCY_OF"
     comment: >-
       Packages for development only. Not part of released artifacts.
+  - pattern: "dokka.*"
+    reason: "BUILD_DEPENDENCY_OF"
+    comment: >-
+      Packages used to build artifacts. Not part of released artifacts.
```
Comment (fviernau) [inline .ort.yml:37]: Consider instead: "Packages for the Dokka documentation engine." 

See also GenerateScopeExcludesCommand.kt.

**Review by fviernau — APPROVED** [review]

**Comment by sschuberth** [comment]
Looks like `funTest-non-analyzer` got stuck again... we probably need to have a look at what's going on there separately.

File: .ort.yml (hunk 1 above)
Comment (sschuberth) [reply .ort.yml:37]: See https://github.com/oss-review-toolkit/ort/pull/6340.

**Pull request merged**